src/BlazorWebView/src/Maui/iOS/IOSWebViewManager.cs

```diff
@@ -53,6 +54,115 @@ internal void MessageReceivedInternal(Uri uri, string message)
 		private void InitializeWebView()
 		{
 			_webview.NavigationDelegate = new WebViewNavigationDelegate(_blazorMauiWebViewHandler);
+			_webview.UIDelegate = new WebViewUIDelegate(_blazorMauiWebViewHandler);
+		}
+
+		internal class WebViewUIDelegate : WKUIDelegate
+		{
+
+			private readonly BlazorWebViewHandler _webView;
+
+			public WebViewUIDelegate(BlazorWebViewHandler webView)
+			{
+				_webView = webView ?? throw new ArgumentNullException(nameof(webView));
+			}
+
+			static string LocalOK = NSBundle.FromIdentifier("com.apple.UIKit").GetLocalizedString("OK");
+			static string LocalCancel = NSBundle.FromIdentifier("com.apple.UIKit").GetLocalizedString("Cancel");
+
+			public override void RunJavaScriptAlertPanel(WKWebView webView, string message, WKFrameInfo frame, Action completionHandler)
+			{
+				PresentAlertController(
+					webView,
+					message,
+					okAction: _ => completionHandler()
+				);
+			}
+
+			public override void RunJavaScriptConfirmPanel(WKWebView webView, string message, WKFrameInfo frame, Action<bool> completionHandler)
+			{
+				PresentAlertController(
+					webView,
+					message,
+					okAction: _ => completionHandler(true),
+					cancelAction: _ => completionHandler(false)
+				);
+			}
+
+			public override void RunJavaScriptTextInputPanel(
+				WKWebView webView, string prompt, string? defaultText, WKFrameInfo frame, Action<string> completionHandler)
+			{
+				PresentAlertController(
+					webView,
+					prompt,
+					defaultText: defaultText,
+					okAction: x => completionHandler(x.TextFields[0].Text!),
+					cancelAction: _ => completionHandler(null!)
+				);
+			}
+
+			static string GetJsAlertTitle(WKWebView webView)
+			{
+				// Emulate the behavior of UIWebView dialogs.
+				// The scheme and host are used unless local html content is what the webview is displaying,
+				// in which case the bundle file name is used.
+				if (webView.Url != null && webView.Url.AbsoluteString != $"file://{NSBundle.MainBundle.BundlePath}/")
```

Comment:
Do you mind linking to a source for this? I'm super new to this space so any pointers would be appreciated.

Reply:
I would suggest looking at what Xamarin.Forms did for its iOS implementation of the regular WebView: https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.iOS/Renderers/WkWebViewRenderer.cs

It looks like it has similar code. That doesn't mean it's what we necessarily want here, because this _isn't_ a regular WebView, but it should be a strong hint about _how_ to do things.